torch/testing/_core.py

```diff
@@ -275,6 +276,111 @@ def make_non_contiguous(tensor: torch.Tensor) -> torch.Tensor:
+    - If :attr:`low` and/or :attr:`high` are specified but exceed the limits: :attr:`dtype` limits are considered instead.
+    - If :attr:`low` is ``-inf`` and/or :attr:`high` is ``inf``: :attr:`dtype` limits are considered instead
+    - If :attr:`low` is ``inf`` or ``nan`` and/or :attr:`high` is ``-inf`` or nan: a `ValueError` is raised, since these are invalid values for the range of output tensor.
+
+    If :attr:`noncontiguous` is ``True``, a noncontiguous tensor with the given size will be returned unless the
+    size specifies a tensor with a 1 or 0 elements in which case the noncontiguous parameter is ignored because
+    it is not possible to create a noncontiguous Tensor with a single element.
+
+    If :attr:`exclude_zero` is ``True`` (default is ``False``), all the values matching to zero in
+    the created tensor are replaced with a ``tiny`` (smallest positive representable number) value if floating type,
+    [``tiny`` + ``tiny``.j] if complex type and ``1`` if integer/boolean type.
+    """
+    def _modify_low_high(low, high, lowest, highest, default_low, default_high, dtype):
+        """
+        Modifies (and raises ValueError when appropriate) low and high values given by the user (input_low, input_high) if required.
+        """
+        def clamp(a, l, h):
+            return min(max(a, l), h)
+
+        low = low if low is not None else default_low
+        high = high if high is not None else default_high
+
+        # Checks for error cases
+        if low != low or high != high:
+            raise ValueError("make_tensor: one of low or high was NaN!")
+        if low > high:
+            raise ValueError("make_tensor: low must be weakly less than high!")
+
+        low = clamp(low, lowest, highest)
+        high = clamp(high, lowest, highest)
+
+        if dtype in integral_types():
+            return math.floor(low), math.ceil(high)
+
+        return low, high
+
+    if dtype is torch.bool:
+        result = torch.randint(0, 2, size, device=device, dtype=dtype)
+    elif dtype is torch.uint8:
+        ranges = (torch.iinfo(dtype).min, torch.iinfo(dtype).max)
+        low, high = _modify_low_high(low, high, ranges[0], ranges[1], 0, 9, dtype)
+        result = torch.randint(low, high, size, device=device, dtype=dtype)
+    elif dtype in integral_types():
+        ranges = (torch.iinfo(dtype).min, torch.iinfo(dtype).max)
+        low, high = _modify_low_high(low, high, ranges[0], ranges[1], -9, 9, dtype)
+        result = torch.randint(low, high, size, device=device, dtype=dtype)
+    elif dtype in floating_types_and(torch.half, torch.bfloat16):
+        ranges_floats = (torch.finfo(dtype).min, torch.finfo(dtype).max)
+        low, high = _modify_low_high(low, high, ranges_floats[0], ranges_floats[1], -9, 9, dtype)
+        rand_val = torch.rand(size, device=device, dtype=dtype)
+        result = high * rand_val + low * (1 - rand_val)
+    else:
+        assert dtype in complex_types()
```

Comment:
We should not have plain `assert`'s in public functions. I suggest you move the complex handling into a separate `elif` branch and raise a `TypeError` in the `else` branch.